lib/platform/gitlab/index.ts

```diff
@@ -582,17 +586,23 @@ export async function updatePr({
       },
     }
   );
+  const pr = res.body;
 
-  await tryPrAutomerge(iid, platformOptions);
+  await tryPrAutomerge(iid, pr, platformOptions);
 }
 
-export async function mergePr(iid: number): Promise<boolean> {
+export async function mergePr(
+  iid: number,
+  _branchName: string,
+  pr: Pr
```

Comment:
```suggestion
```

Do not change the platform api, use new `getMR` function recently added via #9345 to get current `squash` value

Reply:
I've made this change, but it means there's one more API request to GitLab for information renovate already knows.

If you're okay with that, then this PR is ready to re-review @viceice

Reply:
Nope, there won't be a real request, because it uses an http layer cache 😉